<launch>
  <param name="tb3_0/robot_num"  value="0"/>
  <param name="tb3_1/robot_num"  value="1"/>
  <param name="tb3_2/robot_num"  value="2"/>
  <param name="tb3_3/robot_num"  value="3"/>

  <remap from="/robot_num" to="/tb3_0/robot_num"/>
  <remap from="/robot_me" to="/robot_0"/>
  <remap from="/cmd_vel" to="/tb3_0/cmd_vel"/>
  <remap from="/imu" to="/tb3_0/imu"/>
  <remap from="/joint_states" to="/tb3_0/joint_states"/>
  <remap from="/usb_cam/image_raw" to="/tb3_0/usb_cam/image_raw"/>
  <remap from="/usb_cam/image_raw/compressed" to="/tb3_0/usb_cam/image_raw/compressed"/>
  <remap from="/scan" to="/tb3_0/scan"/>
  <remap from="/get_action" to="/tb3_0/get_action"/>
  <remap from="/result" to="/tb3_0/result"/>
  <remap from="/xyzY" to="/tb3_0/xyzY"/>
  <node pkg="turtlebot3_dqn" type="main_goal_4rb" 
  name="main_goal_4rb_0" output="screen"/>

  <remap from="/robot_num" to="/tb3_1/robot_num"/>
  <remap from="/robot_me" to="/robot_1"/>
  <remap from="/cmd_vel" to="/tb3_1/cmd_vel"/>
  <remap from="/imu" to="/tb3_1/imu"/>
  <remap from="/joint_states" to="/tb3_1/joint_states"/>
  <remap from="/usb_cam/image_raw" to="/tb3_1/usb_cam/image_raw"/>
  <remap from="/usb_cam/image_raw/compressed" to="/tb3_1/usb_cam/image_raw/compressed"/>
  <remap from="/scan" to="/tb3_1/scan"/>
  <remap from="/get_action" to="/tb3_1/get_action"/>
  <remap from="/result" to="/tb3_1/result"/>
  <remap from="/xyzY" to="/tb3_1/xyzY"/>
  <node pkg="turtlebot3_dqn" type="main_goal_4rb" 
  name="main_goal_4rb_1" output="screen"/>

  <remap from="/robot_num" to="/tb3_2/robot_num"/>
  <remap from="/robot_me" to="/robot_2"/>
  <remap from="/cmd_vel" to="/tb3_2/cmd_vel"/>
  <remap from="/imu" to="/tb3_2/imu"/>
  <remap from="/joint_states" to="/tb3_2/joint_states"/>
  <remap from="/usb_cam/image_raw" to="/tb3_2/usb_cam/image_raw"/>
  <remap from="/usb_cam/image_raw/compressed" to="/tb3_2/usb_cam/image_raw/compressed"/>
  <remap from="/scan" to="/tb3_2/scan"/>
  <remap from="/get_action" to="/tb3_2/get_action"/>
  <remap from="/result" to="/tb3_2/result"/>
  <remap from="/xyzY" to="/tb3_2/xyzY"/>
  <node pkg="turtlebot3_dqn" type="main_goal_4rb" 
  name="main_goal_4rb_2" output="screen"/>

  <remap from="/robot_num" to="/tb3_3/robot_num"/>
  <remap from="/robot_me" to="/robot_3"/>
  <remap from="/cmd_vel" to="/tb3_3/cmd_vel"/>
  <remap from="/imu" to="/tb3_3/imu"/>
  <remap from="/joint_states" to="/tb3_3/joint_states"/>
  <remap from="/usb_cam/image_raw" to="/tb3_3/usb_cam/image_raw"/>
  <remap from="/usb_cam/image_raw/compressed" to="/tb3_3/usb_cam/image_raw/compressed"/>
  <remap from="/scan" to="/tb3_3/scan"/>
  <remap from="/get_action" to="/tb3_3/get_action"/>
  <remap from="/result" to="/tb3_3/result"/>
  <remap from="/xyzY" to="/tb3_3/xyzY"/>
  <node pkg="turtlebot3_dqn" type="main_goal_4rb" 
  name="main_goal_4rb_3" output="screen"/>

  <node pkg="turtlebot3_dqn" type="optuna_goal_4rb" name="optuna_goal_4rb" output="screen" />


</launch>
